localstack-core/localstack/config.py

```diff
@@ -1077,6 +1077,15 @@ def populate_edge_configuration(
     if not os.environ.get("PROVIDER_OVERRIDE_APIGATEWAYMANAGEMENTAPI"):
         os.environ["PROVIDER_OVERRIDE_APIGATEWAYMANAGEMENTAPI"] = "next_gen"
 
+# Whether the DynamoDBStreams native provider is enabled
+DDB_STREAMS_PROVIDER_V2 = os.environ.get("PROVIDER_OVERRIDE_DYNAMODBSTREAMS", "") == "v2"
+if DDB_STREAMS_PROVIDER_V2:
+    # in order to not have conflicts between the 2 implementations, as they are tightly coupled, we need to set DDB
+    # to be v2 as well
+    # TODO: to be decided if we make DDB Streams the "v2" or DDB, but the biggest change is in the streams themselves
+    if not os.environ.get("PROVIDER_OVERRIDE_DYNAMODB"):
+        os.environ["PROVIDER_OVERRIDE_DYNAMODB"] = "v2"
+
```

Comment:
Should there be a reverse dependency? i.e. if DynamoDB v2 is used, should DynamoDBStreams v2 be used too?

Reply:
I think we could, yes. I thought it'd be easier to communicate about only one flag and the second one being "implicit" in a way, but may be better safe than sorry